app/views/todo_items/_todo_item.html.erb

```diff
@@ -4,8 +4,9 @@
 <%= link_to "Mark as Complete", complete_todo_list_todo_item_path(@todo_list, todo_item.id), method: :patch %>
 -->
 <div class="row clearfix">
+  <% todo_item.assign_status %>
```

Comment:
This would update db.
Your view should never directly interact with the Model -> DB.

Reply:
Additionally implementing this here means you'll be changing priority of a task with respect to its Due Date.
A high priority task can miss a deadline, but that doesn't mean it's priority gets lowered.

Reply:
@achheSharma This is intentional As the priority keeps on increasing as due date comes nearer. So whenever user loads the page. It will get the status based on the current date and due date.

Reply:
and yeah I should keep that method in the controller. agreed.

Reply:
This should be done using a background job, independent of weather or not a user visits the page

Reply:
Yes, It makes sense.